InsightsBackend/src/main/java/org/frankframework/insights/label/LabelService.java

```diff
@@ -85,69 +86,57 @@ public void injectLabels() throws LabelInjectionException {
     }
 
     /**
-     * Fetches labels from the database and returns them as a set of LabelResponse objects.
-     * @param releaseId the ID of the release for which to fetch labels
-     * @return a set of LabelResponse objects
+     * Fetches and processes labels associated with a specific release to determine highlights.
+     * @param releaseId the ID of the release
+     * @return a set of highlighted labels for the release
      * @throws ReleaseNotFoundException if the release is not found
-     * @throws MappingException if an error occurs during mapping
+     * @throws MappingException if there is an error during mapping
      */
     public Set<LabelResponse> getHighlightsByReleaseId(String releaseId)
             throws ReleaseNotFoundException, MappingException {
-        Release release = releaseService.checkIfReleaseExists(releaseId);
-        List<Label> releaseLabels = labelRepository.findLabelsByReleaseId(release.getId());
-        Map<String, Long> labelCountMap = countLabelOccurrences(releaseLabels);
-        Map<String, Label> uniqueLabelMap = mapUniqueLabels(releaseLabels);
+        List<Label> releaseLabels = getLabelsForRelease(releaseId);
+        if (releaseLabels.isEmpty()) {
+            return Collections.emptySet();
+        }
 
-        List<Label> priorityLabelsList = getSortedLabelsByPriorityAndCount(uniqueLabelMap, labelCountMap);
-        List<Label> nonPriorityLabelsList = getSortedLabelsByPriorityAndCount(uniqueLabelMap, labelCountMap);
+        Map<Label, Long> labelCounts = calculateLabelCounts(releaseLabels);
 
-        List<Label> highlightLabels = selectTopLabels(priorityLabelsList, nonPriorityLabelsList);
+        List<Label> highlightLabels = selectFinalHighlights(labelCounts);
```

Comment:
```suggestion
        List<Label> highlightLabels = selectFinalHighlights(releaseLabels);
```

En dan `return calculateLabelCounts(releaseLabels).entrySet().stream()` etc in `selectFinalHighlights`